go.mod

```diff
@@ -4,32 +4,47 @@ go 1.19
 
 require (
 	github.com/ethereum/go-ethereum v1.12.2
+	github.com/iden3/contracts-abi/onchain-credential-status-resolver/go/abi v0.0.0-20230911113809-c58b7e7a69b0
 	github.com/iden3/contracts-abi/rhs-storage/go/abi v0.0.0-20231006141557-7d13ef7e3c48
+	github.com/iden3/contracts-abi/state/go/abi v0.0.0-20230405152923-4a25f6f1f0f4
+	github.com/iden3/go-iden3-core/v2 v2.0.3
 	github.com/iden3/go-merkletree-sql/v2 v2.0.4
-	github.com/stretchr/testify v1.8.2
+	github.com/iden3/go-schema-processor/v2 v2.2.1-0.20240126124145-b6f321093157
+	github.com/jarcoal/httpmock v1.3.1
```

Comment:
What do you mean? You can pass URL to `http/test mock` instance. And return everething that you want. I agree if you have two consecutive calls, it can be problem but you always can manage `http/test mock` with endpoint or domain.